### DISCUSSION TITLE
[discussion] using analytics & crashalytics for logs

### DISCUSSION THREAD

**Discussion by egyleader** [serverpod/serverpod#871] [Q&A] [answered]
Why serverpod took the approach of storing logs & health checks in the project database which would clutter the db and add millions of records overtime & impact db performance while there is free services that is more optimized for that purpose and server independent like firebase analytics & crashalytics ?

**Comment by yahu1031** [comment]
But this is an enhancement AON if you ask me. And moreover, Analytics and Crashalytics have more options(features) in them. So we can hope that feature will also be added soon or later. Something like serverpod_analytics 😈

**Comment by vlidholt** [comment, marked as answer]
The health check take very little space in your database as they are compressed down over time. For the logging, you can specify exactly what to log on a very fine grained level, if you are concerned about the storage space.

The logging is very useful to find slow queries, and text logs are associated with sessions making them much quicker to read through and find relevant issues. Another huge benefit is that you own your data.

Also, there is nothing stopping you from using Crashlytics, but typically that is more for client side logging, no?

**Reply by egyleader** [reply]
thanks ,, it's clear for me now that this is the right approach
